test_files/decorator/decorator.js

```diff
@@ -85,7 +85,7 @@ __decorate([
 ], DecoratorTest.prototype, "x", void 0);
 __decorate([
     decorator,
-    __metadata("design:type", external_1.AClass)
+    __metadata("design:type", Object)
```

Comment:
As discussed offline, we'll need to make sure not to regress here for non-aot mode usage. Try using the update... methods in `transformer_util.ts` instead of the builtin ones, one of them has a fix for this behaviour.